src/relax/op/op_common.cc

```diff
@@ -45,6 +45,10 @@ Array<TensorStructInfo> GetInputTensorStructInfo(const Call& call, const BlockBu
   return input_tensor_sinfo;
 }
 
+template Array<TensorStructInfo> GetInputTensorStructInfo<std::function<const char*(int)>>(
+    const Call& call, const BlockBuilder& ctx, int n_input,
+    std::function<const char*(int)> f_get_input_names);
+
```

Comment:
Added this in order to support lazy name construction, through the approach of lambdas. My only concern is that, I’m not sure if the lambdas will kept inlined in this case, where I instantiated the template function explicitly. I’m kinda doubting that.

Reply:
we can use https://github.com/tlc-pack/relax/blob/relax/include/tvm/ir/op.h#L70 which contains all the information needed

Reply:
if you want to eliminate lambda, you don't want to pass in template specializations, instead make the function an inline function and implcit pass in value by auto